doc/index.md

```diff
@@ -236,114 +264,134 @@ local myconnection = http_client.connect {
+
 
 ### `h1_connection:read_body_by_length(len, timeout)` <!-- --> {#http.h1_connection:read_body_by_length}
 
+Get `len` number of bytes from the socket. Use a negative number for *up to* that number of bytes. This function will yield and wait on the socket if length of the buffered body is less than `len`. Asserts if len is not a number.
+
 
 ### `h1_connection:read_body_till_close(timeout)` <!-- --> {#http.h1_connection:read_body_till_close}
 
+Reads the entire request body. This function will yield until the body is complete or `timeout` is expired. If the read fails the function returns `nil`, an error message and an error number. 
+
 
 ### `h1_connection:read_body_chunk(timeout)` <!-- --> {#http.h1_connection:read_body_chunk}
 
+Reads the next available line of data from the request and returns the chunk and any chunk extensions. This function will yield until chunk size is received or `timeout` is exceeded. If the chunk size is indicated as `0` then `false` and any chunk extensions are returned. Returns `nil`, an error message and an error number if there was an error reading reading the chunk header or the socket.
+
 
 ### `h1_connection:write_request_line(method, path, httpversion, timeout)` <!-- --> {#http.h1_connection:write_request_line}
 
+Sends an HTTP 1.x request line to a remote server, initiating an HTTP request. Yields until success or `timeout`. If the write fails, returns `nil`, an error message and an error number.
+
 
 ### `h1_connection:write_status_line(httpversion, status_code, reason_phrase, timeout)` <!-- --> {#http.h1_connection:write_status_line}
 
+Sends a status response to the remote connection. Yields until success or `timeout`. If the write fails, the funtion returns `nil`, an error message and an error number. 
+
 
 ### `h1_connection:write_header(k, v, timeout)` <!-- --> {#http.h1_connection:write_header}
 
+Writes a header item to the socket as a `key:value` string. Yields until success or `timeout`. Returns `nil`, an error message and an error if the write fails. 
+
 
 ### `h1_connection:write_headers_done(timeout)` <!-- --> {#http.h1_connection:write_headers_done}
 
+Terminates a header block by sending a blank line (`"\r\n"`). Yields until success or `timeout`. Returns `nil`, an error message and an error if the write fails.
+
 
 ### `h1_connection:write_body_chunk(chunk, chunk_ext, timeout)` <!-- --> {#http.h1_connection:write_body_chunk}
+ 
+Writes a line of data to the socket. `chunk_ext` must be `nil` as chunk extensions are not supported. Will yield until complete or `timeout` is exceeded. Returns true on success. Returns `nil`, an error message and an error number if the write fails. 
```

Comment:
It doesn't write a 'line'. it writes the data in `chunk` as a single chunk in the HTTP chunked transfer encoding.